docs/architecture/adr/adr-006-in-cluster-db.md

```diff
@@ -0,0 +1,115 @@
+multiple occasions:
+
+- 17.01.2025: Database went down due to a restart initiated by the Azure Product
+Group team taking unsanctioned action
+- 15.01.2025: Database went down due to a restart initiated by the Azure Product
+Group team taking unsanctioned action
+- 9.01.2025: Database went down as a result of to many login requests
+- 28.11.2024: Database went down due to too many active connection attempts
+- 7.11.2024: Database went down due to too many connections
+- Summer 2024: Database was restarted by Azure for unscheduled and unnotified
+maintenance
+
+Microsoft's database server logs are prohibitively expensive.
+Microsoft misconfigured the database provision-time.
+The current Microsoft support tier is useless and useful support is
+prohibitively expensive.
+Microsoft support has made unsanctioned changes on the databse, such as
+maintenance work during business hours and restarts in attempts to fix support
+tickets.
+It is not possible to access the database server.
+
+Installing a MariaDB Sql operator in the cluster provides the following
+benefits:
+
+- access to the database server logs
+- access to the server itself
+- a guarantee that no one but the platform team can take action on the server
+- ability to right-size the database
+- Running clusters of databases
+- Database recovery
+- The ability to start recovering from a crash immediately
+
+### Pros & Cons
+
+#### Azure Managed DB
+
+##### Cons
+
+- Experienced to be poorly managed
+- Server logs is disproportionately priced = too expensive for this project
+- We don't have access to the server
+- The wait time for support can be long and incorrect.
+- Server might be (was in our case) misconfigured
+- Will require a migration to a potentially as bad service
+
+##### Pros
+
+- One-click setup
+- Minimal configuration needed from us
+- Azure support
+- Azure can be blamed for downtime, they are directly or indirectly responsible
+for.
+- MySql, which is not set to be sunset, might have a noticably better service
+as that is their chosen flavor to keep offering.
+
+#### In-Cluster DB
+
+##### Cons
+
+- We have no one but ourselves to blame if something goes awry
+- Logging and Monitoring has to be set up manually
+- All configuration has to be done ourselves
+- Not possible to do PITR recovery at the moment.
+
+##### Pros
+
+- Ability to investigate logs
+- Logging and monitoring can be set up to alert us to noteworthy changes
+- A little cheaper on the face of it
+- Ability to split out databases such that one database having issues doesn't
+cause all sites to crash.
+- Possible to run in a HA setup if we should decide to do so.
+- Performance can be optimized and finetuned to our usecase
+- Databases will be located closer, logically as well as physically, to
+workloads relying on them = faster response time, which should be noticable for
+end-users.
+- We can right-size the databae, thereby getting the maximum performance for the
+buck.
```

Comment:
As mentioned previously: Personally I do not see money as a key driver here. The database is far from the most expensive service we use.

Reply:
I agree that it isn't the most important driver, but I think it is worth considering along with all the other factors. If running our own database was more expensive, it would definitely be a factor.